File: array_problems/commonInTwoArray/commonInTwoArray.go
package main
import ("fmt")
type ArrayProblems struct{}

func (ap *ArrayProblems) commonInTwoArray(arr1 []int, arr2 []int) []int {
    commons := []int{}
    for _, num := range arr1 {
        if contains(arr2, num) {
            commons = append(commons, num)
        }
    }
    return commons
}
func contains(slice []int, num int) bool {
    for _, n := range slice {
        if n == num {
            return true
        }
    }
    return false
}

func main() {
    arrayProblems := ArrayProblems{}
    arr1 := []int{1, 2, 5, 8, 10, 11, 13, 15}
    arr2 := []int{1, 12, 3, 14, 5, 6, 7, 8, 9, 10}
    found_commons := arrayProblems.commonInTwoArray(arr1, arr2)
    fmt.Println(found_commons)
}

File: array_problems/commonInTwoArray/commonInTwoArray.py
class ArrayProblems:
    def commonsInTwoArray(self, arr1, arr2):
        commons = []
        for i in range(len(arr1)):
            if arr1[i] in arr2:
                commons.append(arr1[i])
        return commons


array_problems = ArrayProblems()
array1 = [1, 12, 3, 14, 5, 6, 7, 8, 9, 10]
array2 = [1, 2, 5, 8, 10, 11, 13, 15]
found_commons = array_problems.commonsInTwoArray(array1, array2)
print(found_commons)
